### PR TITLE
interfaces/greengrass-support: adjust accesses now that have working snap

### PR DESCRIPTION
**Pull request by jdstrand**
- don't use owner match with @{PROC}/[0-9]*/mountinfo
- use @{SNAP_REVISION} with pivot_root rule
- allow 'ro, remount, rbind' on / for pivot_root in overlay

### CODE REVIEW

File: interfaces/builtin/greengrass_support.go [hunk 1]
```diff
@@ -128,11 +128,12 @@ mount options=(rw, bind) /dev/null -> /proc/sched_debug,
 mount options=(rw, bind) /dev/null -> /proc/timer_stats,
 
 # perform the pivot_root into the overlay
-pivot_root oldroot=/var/snap/greengrass/x1/rootfs/.pivot_root*/ /var/snap/greengrass/*/rootfs/,
+pivot_root oldroot=/var/snap/greengrass/@{SNAP_REVISION}/rootfs/.pivot_root*/ /var/snap/greengrass/*/rootfs/,
```
Comment (zyga) [inline interfaces/builtin/greengrass_support.go]: Are you aware of the consequences of using pivot_root and how it makes most of our apparmor rules ineffective?

(Sorry for pulling this out now, I didn't notice this before)

Comment (jdstrand) [inline interfaces/builtin/greengrass_support.go]: greengrass is proprietary software and fails without this access. It relies upon various features to setup its own sandbox.

However, I'm aware of mediation issues regarding overlayfs and pivot root (in particular https://bugs.launchpad.net/apparmor/+bug/1703692), which is just one of many reasons why this interface is a 'super-privileged' interface, like docker-support and lxd-support (ie, you need a snap declaration to even install a snap that plugs this).

Comment (zyga) [inline interfaces/builtin/greengrass_support.go]: Ack, thank you, I just wanted to make sure this is well-known.

**Comment by jdstrand** [comment]
Thanks!

**Pull request merged**